specs/SetMemoryUsageLevel.md

```diff
@@ -0,0 +1,109 @@
+# Background
+Some WebView2 apps want to continue to run scripts while inactive and therefore cannot make usage of `TrySuspend` and `Resume` APIs to reduce resource consumption.
+We are introducing WebView2 API to reduce memory usage only for this type of inactive WebViews.
+
+# Description
+You may call the `SetMemoryUsageLevel` API to have the WebView2 consume less memory and going back to normal usage. This is useful when your Win32 app becomes invisible, 
+but still wants to have script running or monitoring requests from network.
+
+# Examples
+## .NET, WinRT
+```c#
+async protected void OnBecomingInactive()
+{
+    webView.CoreWebView2.SetMemoryUsageLevel(CoreWebView2MemoryUsageLevel.Idle);
+}
+async protected void OnBecomingActive()
+{
+    webView.CoreWebView2.SetMemoryUsageLevel(CoreWebView2MemoryUsageLevel.Normal);
+}
+```
+## Win32 C++
+```cpp
+bool ViewComponent::HandleWindowMessage(
+    HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam, LRESULT* result)
+{
+    if (message == WM_SYSCOMMAND)
+    {
+        if (wParam == SC_MINIMIZE)
+        {
+            OnBecomingInactive();
+        }
+        else if (wParam == SC_RESTORE)
+        {
+            OnBecomingActive();
+        }
+    }
+}
+
+void ViewComponent::OnBecomingInactive()
+{
+    HRESULT hr = webView->SetMemoryUsageLevel(COREWEBVIEW2_MEMORY_USAGE_LEVEL_IDLE);
+    if (FAILED(hr))
+        ShowFailure(hr, L"Failed to set SetMemoryUsageLevel to idle");
+}
+
+void ViewComponent::OnBecomingActive()
+{
+    HRESULT hr = webView->SetMemoryUsageLevel(COREWEBVIEW2_MEMORY_USAGE_LEVEL_NORMAL);
+    if (FAILED(hr))
+        ShowFailure(hr, L"Failed to set SetMemoryUsageLevel to normal");
+}
+
+```
+
+# Remarks
+See [API Details](#api-details) section below for more details.
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+```IDL
+/// Specifies memory usage level of WebView.
+[v1_enum]
+typedef enum COREWEBVIEW2_MEMORY_USAGE_LEVEL {
+    /// Specifies normal memory usage level.
+    COREWEBVIEW2_MEMORY_USAGE_LEVEL_NORMAL,
+
+    /// Specifies idle memory usage level.
```

Comment:
Naming this Idle feels a little weird to me. Idle is naming it after a scenario that can use this rather than naming it after what it does. It sort of seems misleading to me since we're not changing the memory usage to idle. Instead for an idle app scenario you want to change memory usage level to a lower value. What about Reduced or Low?

Or name it to say more what the tradeoff is, like PreferSwapDiskOverInMemory (is that what this does?)

There's an existing enum Windows.System.AppMemoryUsageLevel with values Low, Medium, High, OverLimit. Not totally applicable here.

Reply:
Will change to Low. This is for expressing intent, not used memory. Should we change the name of the API to something like SetMemoryConsumptionTarget or some other name?

Reply:
Yes SetMemoryConsumptionTarget and Low sound good to me!

Reply:
Changed it to MemoryUsageTargetLevel property with Low and Normal values.